### PR TITLE
add better apply section routing

### PR DESCRIPTION
**Pull request by mixcas**

### CODE REVIEW

**Comment by varffs** [comment]
is this not missing `steps.js`?

**Comment by mixcas** [comment]
@varffs no
💭

File: app/imports/components/apply/applyLayout.jsx [hunk 1]
```diff
@@ -13,28 +12,26 @@ export default class ApplyLayout extends Component {
...
-          applicationId={this.props.application._id}
-          step={this.props.section.step}
-          sectionId={this.props.section.id}
-          model={this.props.section.data} />
-      );
-      default: return ':('; // 404
+    let step = this.props.section.step;
+
+    if (step === undefined) {
+      return ':('; // 404
```
Comment (varffs) [inline app/imports/components/apply/applyLayout.jsx]: this doesnt really look like a 404. 404 should be a page component surely!

File: app/imports/components/apply/applyLayout.jsx [hunk 2]
```diff
@@ -13,28 +12,26 @@ export default class ApplyLayout extends Component {
...
     }
 
+    // Steps is an array, so positions start at 0.
+    // But the step numbers for the routs start at 1. So we
+    // Rest the offset to the step param received.
```
Comment (varffs) [inline app/imports/components/apply/applyLayout.jsx]: 'rest' the offset?

**Pull request merged**